venafi/resource_venafi_certificate.go

```diff
@@ -361,9 +361,55 @@ func resourceVenafiCertificateUpdate(_ context.Context, d *schema.ResourceData,
 	return nil
 }
 
-func resourceVenafiCertificateDelete(_ context.Context, d *schema.ResourceData, _ interface{}) diag.Diagnostics {
-	// We don't support deletion for created certificates from TPP, so we just remove it from state
+func resourceVenafiCertificateDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
+
+	provConf := meta.(*venafiProviderConfig)
+	vCertCfg := provConf.vCertCfg
+
+	//VCert connector fake has not implemented the RetireCertificate functionality
+	if !provConf.skipRetirement && vCertCfg.ConnectorType != endpoint.ConnectorTypeFake {
```

Comment:
Change this logic to be positive:

if provConf.skipRetirement{ then do nothing and return}
if vCertCfg.ConnectorType == endpoint.ConnectorTypeFake {
then return and do nothing}

Complexity is being added by having the main logic inside an if condition

Reply:
I understand your point but in that way I'll have to set the `d.SetId("")` three times, something like following:
`if provConf.skipRetirement || vCertCfg.ConnectorType == endpoint.ConnectorTypeFake{
   d.SetId("")
   return nil
}

...
logic to retire certificate
...

d.SetId("")
return nil
`

Reply:
you can set `d.SetId("")` at the beginning of the function, unconditionally.

Reply:
It's better to do just before to return given if something goes wrong, then we don't modify the state.
By other hand I've updated the code following the request approach